test/e2e/logs_test.go

```diff
@@ -35,12 +39,27 @@ func testNodeLogs(t *testing.T) {
 		for _, file := range mandatoryLogs {
 			// A subtest is useful here to attempt to get all the logs and not bail on the first error
 			t.Run(node.Name+"/"+file, func(t *testing.T) {
-				assert.NoError(t, retrieveLog(node.Name, file, nodeDir))
+				err := wait.PollImmediate(retry.Interval, retry.ResourceChangeTimeout, func() (bool, error) {
+					err := retrieveLog(node.GetName(), file, nodeDir)
+					if err != nil {
+						log.Printf("unable to retrieve log %s from node %s: %s", file, node.GetName(), err)
+						return false, nil
+					}
+					return true, nil
+				})
+				assert.NoError(t, err)
 			})
 		}
 		// Grab the optional logs for debugging purposes
 		for _, file := range optionalLogs {
-			_ = retrieveLog(node.Name, file, nodeDir)
+			_ = wait.PollImmediate(retry.Interval, retry.ResourceChangeTimeout, func() (bool, error) {
```

Comment:
Why are we ignoring the error here?

Reply:
These logs are not guaranteed to be present. We still want to collect them if they exist

Reply:
Please add a comment stating this. `Grab the optional logs for debugging purposes` does not explain why we are not looking at the error.